server/src/main/java/org/apache/druid/server/initialization/jetty/JettyBindings.java

```diff
@@ -94,10 +94,15 @@ public Class<? extends Filter> getFilterClass()
     @Override
     public Map<String, String> getInitParameters()
     {
-      if (timeoutMs >= 0) {
-        return ImmutableMap.of("maxRequests", String.valueOf(maxRequests), "suspendMs", String.valueOf(timeoutMs));
+      if (timeoutMs < 0) {
+        return ImmutableMap.of("maxRequests", String.valueOf(maxRequests));
       }
-      return ImmutableMap.of("maxRequests", String.valueOf(maxRequests));
+      if (timeoutMs > Integer.MAX_VALUE) {
+        // QoSFilter tries to parse the suspendMs parameter as an int, so we can't set it to more than Integer
```

Comment:
Since the only use of `timeoutMs` is in this method, maybe we should do the rounding off in the constructor itself.

Reply:
you will still need to execute timeout specific logic. So with the approach you suggested the logic will be scattered across as opposed to being all in one place.

Reply:
Well, it kind of is two separate pieces of logic. One is the max allowed value of `timeoutMs` and the other is construction of the map. I am okay either way though. It is just that the current flow is doing something silently without ever giving any feedback to the user.

I would even suggest just making the `timeoutMs` an `int` instead of a `long`. But if you feel strongly, we can proceed with it as is for now.